File: Boost_Echo_Client/src/AckStatMessage.cpp
//
// Created by MAY on 05/01/2022.
//

#include "../include/AckStatMessage.h"
/*REGISTER E 1 11-11-1111
LOGIN E 1 1
REGISTER S 1 11-11-1111
REGISTER D 1 11-11-1111
REGISTER A 1 01-12-2002
STAT A|D|S

REGISTER E 1 11-11-1111
LOGIN E 1 1
REGISTER S 1 11-11-1111
REGISTER D 1 11-11-1111
REGISTER A 1 01-12-2002
STAT A|D|S
19 0 0 0
91 0 0 0
BLOCK S
ACK 8
STAT A|D|S
 */

AckStatMessage::AckStatMessage(std::list<AckMessageUserInfo> _userInfos, short _messageOpcode, short _opCode) : AckMessage(_messageOpcode, _opCode), userInfos(_userInfos) {
}

std::string AckStatMessage::toString() {
    std::string answer="ACK ";
    answer.append(std::to_string(getMessageOpCode()));
    answer.push_back(' ');
    for(AckMessageUserInfo messageUserInfo : userInfos){
        answer.push_back('\n');
        answer.append(messageUserInfo.toString());
    }
    return answer;
}

File: spl-net/src/main/java/bgu/spl/net/impl/messages/AckStatMessage.java
package bgu.spl.net.impl.messages;

import java.util.LinkedList;

public class AckStatMessage extends AckMessage {
    private LinkedList<AckMessageUserInfo> userInfos;
    public AckStatMessage(short messageOpcode, LinkedList<AckMessageUserInfo> userInfos) {
        super(messageOpcode);
        this.userInfos=userInfos;
    }

    public LinkedList<AckMessageUserInfo> getUserInfos(){return userInfos;}
    @Override
    public String toString(){
        short opcode=getOpCode();
        String output=String.valueOf(opcode);
        short messageOpcode=getMessageOpcode();
        output+=messageOpcode;

        LinkedList<AckMessageUserInfo> userinfo=getUserInfos();
        String info="";
        for(int i=0;i<userinfo.size();i++){
            info+=userinfo.get(i).getAge();
            info+=userinfo.get(i).getNumPosts();
            info+=userinfo.get(i).getNumFollowers();
            info+=userinfo.get(i).getNumFollowing();
            info+="\\|";
        }
        output+=info;

        return output;
    }
}